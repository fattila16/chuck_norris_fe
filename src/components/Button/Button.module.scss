.button {
    font-family: inherit;
    border: 0;
    box-sizing: border-box;
    padding: 10px;
    cursor: pointer;
    white-space: nowrap;
    text-decoration: none;
    width: 100%;
    height: 100%;
    display: inline-flex;
    align-items: center;
    justify-content: center;
    border-radius: 10px;
    box-shadow: 0 3px 5px rgba(0, 0, 0, 0.18);
    text-align: center;
    line-height: 1.1;
    background-color: #606c38;
    &:hover {
        transform: translateY(-1px);
        box-shadow:0px 4px 18px rgba(0,0,0,0.2);
    }
}

.medium {
    width: 20ch;
    height: 44px;
}