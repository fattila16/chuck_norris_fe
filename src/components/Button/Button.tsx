import React, { MouseEvent } from 'react';
import css from './Button.module.scss';

interface ButtonProps {
    onClick?: (event: MouseEvent<HTMLButtonElement>) => void;
    title?: string;
    size?: 'small' | 'medium' | 'big';
    id?: string;
}
const Button: React.FC<ButtonProps> = (props: ButtonProps) => {
    const { title, onClick, size = 'medium', id } = props;
    return (
        <button id={id} className={`${css.button} ${css[size]}`} onClick={onClick}>
            {title}
        </button>
    );
};

export default Button;
