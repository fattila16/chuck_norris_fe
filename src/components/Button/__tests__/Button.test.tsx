import React from 'react';
import { shallow } from 'enzyme';
import Button from '../Button';
import sinon from 'sinon';

describe('[Button]', () => {
    it('Should render Button component without error', () => {
        const component = shallow(<Button />);
        expect(component.getElements()).toMatchSnapshot();
    });

    it('Should render out title from props', () => {
        const component = shallow(<Button title="Test" />);
        expect(component.render().text()).toMatch('Test');
    });

    it('Should render button and call the onClick prop function when clicked', () => {
        const onClickSpy = sinon.spy();
        const component = shallow(<Button onClick={onClickSpy} />);
        component.simulate('click');
        sinon.assert.called(onClickSpy);
    });
});
