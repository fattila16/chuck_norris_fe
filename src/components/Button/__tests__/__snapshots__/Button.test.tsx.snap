// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Button] Should render Button component without error 1`] = `
Array [
  <button
    className="button medium"
  />,
]
`;
